api/src/main/java/org/apache/iceberg/metrics/LongCounter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.metrics;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * A default {@link org.apache.iceberg.metrics.MetricsContext.Counter} implementation that uses a {@link Long} to count
+ * events.
+ */
+class LongCounter implements MetricsContext.Counter<Long> {
+  private final AtomicLong counter;
+
+  LongCounter() {
+    counter = new AtomicLong(0L);
+  }
+
+  @Override
+  public void increment() {
+    counter.incrementAndGet();
+  }
+
+  @Override
+  public void increment(Long amount) {
+    counter.addAndGet(amount);
+  }
+
+  @Override
+  public Optional<Long> count() {
+    return Optional.of(counter.get());
+  }
+
+  @Override
+  public String toString() {
+    return count().orElse(-1L).toString();
```

Comment:
This will never be empty, right? Why not just print the value of counter directly?

Reply:
makes sense, updated